plugins/woocommerce-blocks/assets/js/blocks/checkout/inner-blocks/checkout-additional-information-block/block.tsx

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+import { noticeContexts } from '@woocommerce/base-context';
+import { __ } from '@wordpress/i18n';
+import { StoreNoticesContainer } from '@woocommerce/blocks-components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { CHECKOUT_STORE_KEY } from '@woocommerce/block-data';
+import { ADDITIONAL_FORM_KEYS } from '@woocommerce/block-settings';
+import { Form } from '@woocommerce/base-components/cart-checkout';
+import type { FunctionComponent } from 'react';
+import NoticeBanner from '@woocommerce/base-components/notice-banner';
+
+const Block: FunctionComponent = () => {
+	const { additionalFields, isEditor } = useSelect( ( select ) => {
+		const store = select( CHECKOUT_STORE_KEY );
+		const editorStore = select( 'core/editor' );
+		return {
+			additionalFields: store.getAdditionalFields(),
+			isEditor: !! editorStore,
+		};
+	} );
+
+	const { setAdditionalFields } = useDispatch( CHECKOUT_STORE_KEY );
+
+	const onChangeForm = ( additionalValues ) => {
+		setAdditionalFields( additionalValues );
+	};
+
+	const additionalFieldValues = {
+		...additionalFields,
+	};
+
+	if ( ADDITIONAL_FORM_KEYS.length === 0 && ! isEditor ) {
+		return null;
+	}
+
+	if ( ADDITIONAL_FORM_KEYS.length === 0 && isEditor ) {
+		return (
+			<NoticeBanner status="warning" isDismissible={ false }>
+				{ __(
+					'There are no custom fields registered. This block will not be visible on the front-end.',
+					'woocommerce'
+				) }
+			</NoticeBanner>
+		);
+	}
```

Comment:
Do we want to show a warning? How about show nothing? (hidden)

Reply:
Hmm, yeah actually this would be a noisy experience for the merchant if they never intend to add custom fields. I'll hide it.